src/Component/RuleGenerator/RuleGenerator.tsx

```diff
@@ -0,0 +1,398 @@
+  quantile: string;
+}
+
+// default props
+interface RuleGeneratorDefaultProps {
+  unknownSymbolizerText?: string;
+  locale: RuleGeneratorLocale;
+  classificationMethods: ClassificationMethod[];
+  colorRampNames: string[];
+}
+
+interface RuleGeneratorState {
+  attributeName?: string;
+  attributeType?: string;
+  numberOfRules?: number;
+  levelOfMeasurement?: LevelOfMeasurement;
+  colorRamp?: string[];
+  colorRampName?: string;
+  symbolizerKind?: SymbolizerKind;
+  wellKnownName?: WellKnownName;
+  classificationMethod?: ClassificationMethod;
+}
+
+// non default props
+export interface RuleGeneratorProps extends Partial<RuleGeneratorDefaultProps> {
+  rules: Rule[];
+  internalDataDef: Data;
+  onRulesChange?: (rules: Rule[]) => void;
+}
+
+export class RuleGenerator extends React.Component<RuleGeneratorProps, RuleGeneratorState> {
+
+  public static defaultProps: RuleGeneratorDefaultProps = {
+    locale: en_US.GsRuleGenerator,
+    classificationMethods: ['equalInterval', 'quantile'],
+    colorRampNames: [
+      'jet', 'hsv', 'hot', 'cool', 'spring', 'summer', 'autumn', 'winter', 'bone',
+      'copper', 'greys', 'YIGnBu', 'greens', 'YIOrRd', 'bluered', 'RdBu', 'picnic',
+      'rainbow', 'portland', 'blackbody', 'earth', 'electric',
+      'viridis', 'inferno', 'magma', 'plasma', 'warm', 'rainbow-soft',
+      'bathymetry', 'cdom', 'chlorophyll', 'density', 'freesurface-blue',
+      'freesurface-red', 'oxygen', 'par', 'phase', 'salinity', 'temperature',
+      'turbidity', 'velocity-blue', 'velocity-green'
+    ]
+  };
+
+  constructor(props: RuleGeneratorProps) {
+    super(props);
+
+    const {
+      internalDataDef
+    } = this.props;
+
+    const symbolizerKind = RuleGeneratorUtil.guessSymbolizerFromData(internalDataDef);
+
+    this.state = {
+      numberOfRules: 2,
+      symbolizerKind
+    };
+  }
+
+  onAttributeChange = (attributeName: string) => {
+    const {
+      internalDataDef
+    } = this.props;
+    const attributeType = _get(internalDataDef, `schema.properties[${attributeName}].type`);
+    let numberOfRules: number;
+    if (attributeType === 'string') {
+      const distinctValues = RuleGeneratorUtil.getDistinctValues(internalDataDef, attributeName);
+      numberOfRules = distinctValues.length;
+    }
+    this.setState({
+      attributeName,
+      attributeType,
+      levelOfMeasurement: attributeType === 'string' ? 'nominal' : 'cardinal',
+      numberOfRules: numberOfRules || this.state.numberOfRules
+    });
+  }
+
+  onLevelOfMeasurementChange = (event: RadioChangeEvent) => {
+    const levelOfMeasurement = event.target.value;
+    this.setState({levelOfMeasurement});
+  }
+
+  onClassificationChange = (classificationMethod: ClassificationMethod) => {
+    this.setState({classificationMethod});
+  }
+
+  onNumberChange = (numberOfRules: number) => {
+    let {
+      colorRamp,
+      colorRampName
+    } = this.state;
+
+    if (numberOfRules <= 2) {
+      colorRamp = undefined;
```

Comment:
this.state is readonly. `colorRamp` and `colorRampName` should be cloned before being assigned a new value